### PR TITLE
Add the 0.24.0 release artifacts to the main branch

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Task

### Description

Since 0.24.0 was released, we should add the release artifacts to the `main` branch. This PR adds:
* Helm Chart index
* Helm Chart, examples and install files
* Updates the next product version in docs attributes
* etc.

### CODE REVIEW

File: examples/metrics/grafana-dashboards/strimzi-kafka-bridge.json [hunk 1]
```diff
@@ -1956,7 +1956,7 @@
       "steppedLine": false,
       "targets": [
         {
-          "expr": "sum(rate(strimzi_bridge_http_server_requests_total{container=~\"^.+-bridge\"}[1m])) by (method)",
+          "expr": "sum(rate(strimzi_bridge_http_server_requestCount_total{container=~\"^.+-bridge\"}[1m])) by (method)",
```
Comment (LiamClarkeNZ) [inline examples/metrics/grafana-dashboards/strimzi-kafka-bridge.json:1959]: Wondering why this metric name changed?

The original metric name is aligned to [Prometheus best practices](https://prometheus.io/docs/practices/naming/), the new one, less so.

Comment (sknot-rh) [reply examples/metrics/grafana-dashboards/strimzi-kafka-bridge.json]: The metric names have changed due to vertx update (https://github.com/vert-x3/wiki/wiki/4.0.0-Deprecations-and-breaking-changes#%EF%B8%8F-full-review-of-metric-names)
However, this change is probably a typo.

**Review by sknot-rh — APPROVED** [review]

File: examples/metrics/grafana-dashboards/strimzi-kafka-bridge.json (hunk 1 above)
Comment (scholzj) [reply examples/metrics/grafana-dashboards/strimzi-kafka-bridge.json]: The main point really is that we screwed up in the original PR changed it in the wrong place. It should be fixed now.

Comment (scholzj) [reply examples/metrics/grafana-dashboards/strimzi-kafka-bridge.json]: In any case good catch @LiamClarkeNZ!

Comment (LiamClarkeNZ) [reply examples/metrics/grafana-dashboards/strimzi-kafka-bridge.json]: :+1:

**Pull request merged**